docs/DeveloperGuide.md

```diff
@@ -1,37 +1,1040 @@
+> have been removed from the sequence diagram. Reference frames will be elaborated further
+> down this section.
+
+![Update Schedule Sequence Diagram](uml/sequenceDiagrams/schedule/images/updateSchedule.png)
+<br><br>
+
+(Steps 1 to 3) The program waits for the user's input, which in this case,
+is the schedule `/update <day number> <plan number>` command.
+An example of a valid command would be `schedule /update 1 1`.
+Once the command is entered, the UI class will return the user input in a String object to the WerkIt object.
+
+(Steps 4 to 5) After the user input is received, the WerkIt object will call the `Parser#parseUserInput()` method
+to parse the user input. A return `ScheduleCommand` object is obtained from the method.
+This `ScheduleCommand` object is upcasted to a `Command` object on return to the WerkIt object.
+
+(Step 6) The schedule command is being executed by calling the `ScheduleCommand#execute()` method.
+
+Steps 7 and 8 are explained in greater details in the following sequence diagram:
+
+![updateScheduleDetails](uml/sequenceDiagrams/schedule/images/updateScheduleDetails.png)
+
+(Steps 7.1 to 7.2) The DayList#updateDay method will be called to update/add a plan for a particular day in the schedule 
+stated by the user. It will fist call the String#split method to separate out the `userArgument` given by the user. Upon, 
+splitting of the whitespaces in `userArgument`, it will then check if the `userArgument` is valid. If it is invalid, an 
+Exception would be thrown to the user and following the termination of the process (step 7.4).
+
+(Steps 7.6 to 7.8) After splitting and checking the validity of `userArgument`, variables `userArgument[0]` representing
+day number and `userArgument[1]` representing the plan index of the plan stored in the planList are obtained. Both the 
+variables are then converted from data type string to integer. In addition, there is a check executed on both the converted 
+day number and plan number to ensure that they are valid. This check is done so by calling the `DayList#isDayValid` and
+`DayList#isPlanValid` methods respectively.
+
+(Steps 7.9 and 7.11) If the day number or plan number is not valid, an `InvalidScheduleException` would be thrown to the user,
+and the entire updating of a plan for a particular day in the schedule process is aborted.
+
+(Steps 7.13 and 7.14) `PlanList#getPlanDisplayList` method is called to find and return the hash value of the plan number, 
+`planToAddKey:String` to be scheduled for a particular day. The `planToAddKey` is used to get the `plan` object in the 
+planList by calling the `PlanList#getPlanFromKey` (steps 7.15 to 7.16).
+
+(Steps 7.17 to 7.18) Once the `Plan` object is retrieved, if there are no plan being scheduled for the day, 
+a new Day object is created and stored in the application.
+
+(Steps 7.19 to 7.20) However, if there is an existing plan scheduled for that particular day, the `Day` object that 
```

Comment:
Considering there are a lot of sub-steps here, perhaps you could break it down to more steps?